plugins/woocommerce-admin/client/products/fields/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -91,100 +100,155 @@ export const AttributeTermInputField: React.FC<
+				placeholder={ placeholder || '' }
+				getItemLabel={ ( item ) => item?.name || '' }
+				getItemValue={ ( item ) => item?.slug || '' }
+				stateReducer={ ( state, actionAndChanges ) => {
+					const { changes, type } = actionAndChanges;
+					switch ( type ) {
+						case selectControlStateChangeTypes.ControlledPropUpdatedSelectedItem:
+							return {
+								...changes,
+								inputValue: state.inputValue,
+							};
+						case selectControlStateChangeTypes.ItemClick:
+							if (
+								changes.selectedItem &&
+								changes.selectedItem.id === -99
+							) {
+								return changes;
+							}
+							return {
+								...changes,
+								isOpen: true,
+								inputValue: state.inputValue,
+								highlightedIndex: state.highlightedIndex,
+							};
+						default:
+							return changes;
+					}
+				} }
+				selected={ value }
+				onSelect={ onSelect }
+				onRemove={ onRemove }
+				className="woocommerce-attribute-term-field"
+			>
+				{ ( {
+					items,
+					highlightedIndex,
+					getItemProps,
+					getMenuProps,
+					isOpen,
+				} ) => {
+					return (
+						<Menu isOpen={ isOpen } getMenuProps={ getMenuProps }>
+							{ [
+								isFetching ? (
+									<div
+										key="loading-spinner"
+										className="woocommerce-attribute-term-field__loading-spinner"
 									>
-										<>
-											<CheckboxControl
-												onChange={ () => null }
-												checked={ isSelected }
-												label={
-													<span
-														style={ {
-															fontWeight:
-																isSelected
-																	? 'bold'
-																	: 'normal',
-														} }
-													>
-														{ item.name }
+										<Spinner />
+									</div>
+								) : null,
+								...items.map( ( item, menuIndex ) => {
+									const isSelected =
+										selectedTermSlugs.includes( item.slug );
+
+									return (
+										<MenuItem
+											key={ `${ item.slug }` }
+											index={ menuIndex }
+											isActive={
+												highlightedIndex === menuIndex
+											}
+											item={ item }
+											getItemProps={ getItemProps }
+										>
+											{ item.id !== -99 ? (
+												<CheckboxControl
+													onChange={ () => null }
+													checked={ isSelected }
+													label={
+														<span
+															style={ {
+																fontWeight:
+																	isSelected
+																		? 'bold'
+																		: 'normal',
+															} }
+														>
+															{ item.name }
+														</span>
+													}
+												/>
+											) : (
+												<div className="woocommerce-attribute-term-field__add-new">
+													<Icon
+														icon={ plus }
+														size={ 20 }
+														className="woocommerce-attribute-term-field__add-new-icon"
+													/>
+													<span>
+														{ sprintf(
+															/* translators: The name of the new attribute to be created */
```

Comment:
Just a note that this is the name of the attribute *term, not attribute, in case it matters to the translators.

Reply:
Updated